tests/MockDb.h

```diff
@@ -244,6 +245,22 @@ struct MockDbT : public sqlpp::connection
   {
     return {name};
   }
+
+  void start_transaction(sqlpp::isolation_level level)
+  {
+    _current_isolation_level = level;
```

Comment:
Yes, that does not look nice, but I think this test is useful to verify that the correct internal start_transaction method gets called with the expected level.

I added an extra "InternalMockData" and moved the isolation_level there to make it explicit that this is only for testing.

Does this look better?